warp/src/main/java/com/schibsted/nmp/warp/components/WarpRadio.kt

```diff
@@ -0,0 +1,337 @@
+        enabled: Boolean,
+        isError: Boolean,
+        interacted: Boolean
+    ): State<Color>? {
+        return when {
+            !enabled -> rememberUpdatedState(disabledInteriorColor)
+            interacted && !isError -> rememberUpdatedState(pressedDefaultInteriorColor)
+            interacted && isError -> rememberUpdatedState(pressedErrorInteriorColor)
+            else -> null
+        }
+    }
+}
+
+@Composable
+internal fun defaultRadioColors() = WarpRadioButtonColors(
+    unselectedColor = colors.border.default,
+    selectedColor = colors.border.selected,
+    pressedColor = colors.border.primary,
+    disabledColor = colors.border.disabled,
+    errorColor = colors.border.negative,
+    disabledInteriorColor = colors.background.disabledSubtle,
+    pressedDefaultInteriorColor = colors.background.hover,
+    pressedErrorInteriorColor = colors.background.negativeSubtle,
+)
+
+
+internal object RadioButtonTokens {
+    const val RadioAnimationDuration = 100
+}
+
+/**
+ * A vertical group of Radio buttons in the warp design system.
+ * For more info, look [here](https://warp-ds.github.io/tech-docs/components/radio/)
+ *
+ * @param modifier The modifier to be applied to the radio group.
+ * @param title The title of the radio group. Default value is null.
+ * @param options The list of options for the radio group
+ * @param selectedOption The currently selected option in the radio group
+ * @param helpText An optional help text that will be displayed below the radio group. Null by default
+ * @param enabled Set to false to disable the radio group. Default value is true
+ * @param isError Set to true if the radio group is in an error state. Default value is false
+ * @param onOptionSelected A function that will be invoked when an option in the radio group is selected. Make sure to change the state of currently selected option
+ */
+@Composable
+fun VerticalWarpRadioGroup(
+    modifier: Modifier = Modifier,
+    title: String? = null,
+    options: List<String>,
+    selectedOption: String? = null,
+    helpText: String? = null,
+    enabled: Boolean = true,
+    isError: Boolean = false,
+    onOptionSelected: (String) -> Unit
+) {
+    require(options.isNotEmpty()) { "RadioGroup must have more than 1 element" }
+    // Modifier.selectableGroup() is essential to ensure correct accessibility behavior
+    Column(modifier.selectableGroup(), verticalArrangement = Arrangement.spacedBy(dimensions.space2)) {
+        title?.let {
+            WarpText(
+                text = title,
+                style = WarpTextStyle.Title5,
+                modifier = Modifier.semantics { heading() })
+        }
+        options.forEach { text ->
+            WarpRadio(
+                text = text,
+                onClick = { onOptionSelected(text) },
+                selected = (text == selectedOption),
+                enabled = enabled,
+                isError = isError,
+            )
+        }
+
+        val helpTextColor =
```

Comment:
helpTextColor could be a separate function that returns `: State<Color>` since it's used more than once